src/Framework/AspNetCoreAnalyzers/test/WebApplicationBuilder/UseCreateHostBuilderInsteadOfCreateWebHostBuilderTest.cs

```diff
@@ -0,0 +1,400 @@
+        };
+
+        // No diagnostics expected
+        await VerifyAnalyzer.VerifyAnalyzerAsync(source, expectedDiagnostics);
+    }
+
+    [Fact]
+    public async Task WarnsForPrivateMethods()
+    {
+        // Arrange
+        var source = @"
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore;
+public static class Program
+{
+    public static void Main(string[] args)
+    {
+        CreateWebHostBuilder(args).Build().Run();
+    }
+
+    private static {|#0:IWebHostBuilder|} CreateWebHostBuilder(string[] args) =>
+        {|#1:WebHost.CreateDefaultBuilder(args)|}.UseStartup<Startup>();
+}
+public class Startup { }
+";
+
+        var expectedDiagnostics = new[]
+        {
+            VerifyAnalyzer.Diagnostic().WithLocation(0),
+            VerifyAnalyzer.Diagnostic().WithLocation(1)
+        };
+
+        // No diagnostics expected
+        await VerifyAnalyzer.VerifyAnalyzerAsync(source, expectedDiagnostics);
+    }
+
+    [Fact]
+    public async Task CodeFixWorksInsideUsingStatement()
+    {
+        // Arrange
+        var source = @"
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore;
+public static class Program
+{
+    public static void Main(string[] args)
+    {
+        using (var host = {|#0:WebHost.CreateDefaultBuilder(args)|}
+                .UseStartup<Startup>()
+                .Build())
+        {
+            host.Run();
+        }
+    }
+}
+public class Startup { }
+";
+
+        var fixedSource = @"
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore;
+using Microsoft.Extensions.Hosting;
+
+public static class Program
+{
+    public static void Main(string[] args)
+    {
+        using (var host = Host.CreateDefaultBuilder(args)
+.ConfigureWebHostDefaults(webBuilder => webBuilder.UseStartup<Startup>()
+).Build())
+        {
+            host.Run();
+        }
+    }
+}
+public class Startup { }
+";
+
+        var diagnostic = new DiagnosticResult(DiagnosticDescriptors.UseCreateHostBuilderInsteadOfCreateWebHostBuilder)
+            .WithMessage(Resources.Analyzer_UseCreateHostBuilderInsteadOfCreateWebHostBuilder_Message);
+
+        var expectedDiagnostics = new[]
+        {
+            diagnostic.WithLocation(0),
+        };
+
+        // Assert
+        await VerifyCS.VerifyCodeFixAsync(source, expectedDiagnostics, fixedSource);
+    }
+
+    [Fact]
+    public async Task CodeFixWorksWithManyChainedCalls()
+    {
+        // Arrange
+        var source = @"
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore;
+public static class Program
+{
+    public static void Main(string[] args)
+    {
+        {|#0:WebHost.CreateDefaultBuilder(new[] { ""--cliKey"", ""cliValue"" })|}
+            .ConfigureServices((context, service) => { })
+            .ConfigureKestrel(options =>
+                options.Configure(options.ConfigurationLoader.Configuration))
+            .Configure(app =>
+            {
+            });
+    }
+}
+";
+
+        var fixedSource = @"
```

Comment:
@CyrusNajmabadi wonder if you have any insight into making the formatting a bit nicer here? See fixed code example below not having 'nice' indentation like the original code above.